### PR TITLE
[React] Add Traffic Light challenge

### PR DESCRIPTION
**Pull request by rishabhm05**
# Title and Issue number 

Title : TrafficLightChallenge

Issue No. :

Code Stack :  React JS

Close #<issue_no>

# Checklist:

- [] I have mentioned the issue number in my Pull Request.
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have created a helpful and easy to understand `README.md`
- [] I have updated the Index.html file for my contribution

### CODE REVIEW

File: react/src/helpers/challenges.ts [hunk 1]
```diff
@@ -445,4 +445,14 @@ export const challenges = new Map<string, Challenge>([
...
+  [
+    'traffic-lights',
+    {
+      title: 'Traffic Lights',
+      link: 'traffic-lights',
+      difficulty: 'medium',
+      developer: 'rishabhm05',
+      tags: [],
+    },
+  ],
```
Comment (arpansaha13) [inline react/src/helpers/challenges.ts:457]: Put the new challenge along with the other medium difficulty challenges

**Review by arpansaha13 — CHANGES_REQUESTED** [review]
Consider showing a timer for how long the red and green lights will show

File: react/src/machine-coding/traffic-light/App.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    <div className={styles.AppContainer}>
+      <div className={styles.trafficWrapper}>
+        {Object.keys(config).map((light, index) => {
+          return (
+            <div
+              className={styles.light}
+              style={{ background: `${currentActiveLight === light ? `${light}` : ''}` }}
+            ></div>
+          );
+        })}
```
Comment (arpansaha13) [inline react/src/machine-coding/traffic-light/App.jsx:24]: Each item in list-rendering should have a unique `key` prop. Please add the `key` prop in the list elements.

Note that the `index` variable is unused. Please remove it if it is not required. Also, ideally, index should not be used as key.

**Review by arpansaha13 — CHANGES_REQUESTED** [review]

**Comment by rishabhm05** [comment]
@arpansaha13  Hi  i have done required changes
1 . Added key for each light item
2. Displayed duration for Active Light
3 . Added challenge below the other medium challenges

**Review by arpansaha13 — APPROVED** [review]

**Comment by arpansaha13** [comment]
Thank you for your contribution!

**Comment by arpansaha13** [comment]
I have noticed a bug right now. Check out this issue #325.
Would you like to take it up?

**Comment by rishabhm05** [comment]
yes @arpansaha13  Please assign it to me.

**Comment by arpansaha13** [comment]
I am not able to add you as an assignee.
But you can open the PR for it anyway 👍

**Comment by rishabhm05** [comment]
okay sure

**Pull request merged**